src/components/user/services/userProfileSrv.js

```diff
@@ -1,76 +1,83 @@
-                        var emailFromAuth = authData.password ? authData.password.email : '';
-                        var nickNameFromAuth = authData.auth ? authData.auth.name : emailFromAuth;
+            var authData = AuthService.getAuth();
+            var profilePath = 'users/' + authData.uid + '/profile';
+            return UserStorageService.get(profilePath).then(function (profile) {
+                if (profile && (angular.isDefined(profile.email) || angular.isDefined(profile.nickname))) {
+                    return profile;
+                } else {
+                    var emailFromAuth = authData.auth ? authData.auth.email : authData.password ? authData.password.email : '';
+                    var nickNameFromAuth = authData.auth.name ? authData.auth.name : nickNameFromEmail(emailFromAuth);
 
-                        if (!profile.email) {
-                            profile.email = emailFromAuth;
-                        }
-                        if (!profile.nickname) {
-                            profile.nickname = nickNameFromAuth;
-                        }
-                        if (!profile.createdTime) {
-                            profile.createdTime = StorageSrv.variables.currTimeStamp;
-                        }
+                    if (!profile) {
+                        profile = {
+                            email: emailFromAuth,
+                            nickname: nickNameFromAuth,
+                            createdTime: Firebase.ServerValue.TIMESTAMP
+                        };
+                    }
+                    if (!profile.email) {
+                        profile.email = emailFromAuth;
+                    }
+                    if (!profile.nickname) {
+                        profile.nickname = nickNameFromAuth;
+                    }
+                    if (!profile.createdTime) {
+                        profile.createdTime = Firebase.ServerValue.TIMESTAMP;
+                    }
 
-                        return globalStorage.set(profilePath, profile);
-                    });
-                });
+                    _this.setProfile(profile);
+                    return profile;
+                }
             });
         };
 
         this.getProfileByUserId = function (userId) {
             var userProfilePath = 'users/' + userId + '/profile';
-            return InfraConfigSrv.getGlobalStorage().then(function(globalStorage) {
-                return globalStorage.get(userProfilePath);
-            });
+            return UserStorageService.get(userProfilePath);
         };
 
         this.setProfile = function (newProfile) {
-            return InfraConfigSrv.getGlobalStorage().then(function(globalStorage) {
-                return globalStorage.set(profilePath, newProfile);
+            var authData = AuthService.getAuth();
+            var profilePath = 'users/' + authData.uid + '/profile';
+            return UserStorageService.get(profilePath).then(function (profile) {
```

Comment:
when you call setProfile you alreadu have the uid ? if so maybe send it as param instead of get it